src/components/Header/HeaderSearch.tsx

```diff
@@ -0,0 +1,48 @@
+import { SearchIcon } from "@chakra-ui/icons";
+import { Box, IconButton, useDisclosure } from "@chakra-ui/react";
+import type { FunctionComponent } from "react";
+import { useLocation } from "react-router-dom";
+import { ROUTES } from "../../constants/url";
+import { SearchBar, SearchOverlay, SearchSuggestions } from "../SearchBar";
+import { SearchModal } from "../SearchModal";
+import testIds from "./testIds";
+
+/**
+ * Renders a SearchBar at desktop resolutions and shows a search icon which opens a modal on mobile resolutions
+ */
+export const HeaderSearch: FunctionComponent = () => {
+  const searchModal = useDisclosure();
+  const { pathname } = useLocation();
+
+  if ([ROUTES.HOME, ROUTES.SEARCH].some((path) => path === pathname)) {
+    return null;
+  }
```

Comment:
Can't we just use react-router `Route` here normally to not render on home/search?

```jsx
  <Switch>
    <Route path={[ROUTES.HOME, ROUTES.Search]} />
    <Route path="*">
      <>{mySuperComponent}</>
    </Route>
  </Switch>
```

Reply:
Yeah we totally could, this is just the same code from the former `SearchButton.tsx` did. But I do think the react-router suggestion is a good one